crates/markdown_preview/src/markdown_preview_view.rs

```diff
@@ -119,16 +105,100 @@ impl MarkdownPreviewView {
                 },
             );
 
-            Self {
+            let mut this = Self {
                 selected_block: 0,
                 focus_handle: cx.focus_handle(),
-                workspace,
+                active_editor: None,
+                workspace: workspace.clone(),
                 contents,
                 list_state,
+            };
+
+            this.set_editor(active_editor, cx);
+
+            if let Some(workspace) = &workspace.upgrade() {
+                cx.observe(workspace, |this, workspace, cx| {
+                    let item = workspace.read(cx).active_item(cx);
+                    this.workspace_updated(item, cx);
+                })
+                .detach();
+            } else {
+                log::error!("Failed to listen to workspace updates");
             }
+
+            this
         })
     }
 
+    fn workspace_updated(
+        &mut self,
+        active_item: Option<Box<dyn ItemHandle>>,
+        cx: &mut ViewContext<Self>,
+    ) {
+        if let Some(item) = active_item {
+            if item.item_id() != cx.entity_id() {
+                if let Some(editor) = item.act_as::<Editor>(cx) {
```

Comment:
> *This implementation does not yet discriminate on file type.

Let's make it, should be simple: here, we can do

`editor.read(cx).buffer().read(cx).language_at(0, cx)` and actually check that it's actually a markdown one.

Otherwise we get some really odd things going on:
![image](https://github.com/zed-industries/zed/assets/2690773/62378fed-c01e-4d3f-b8e0-f1cd698afa1c)

This should already be enough to react on file language changes, so we should be good.
I guess we could render a blank `div()` for active items that do not match our expectations.